integration-test/test.sh

```diff
@@ -0,0 +1,350 @@
+#!/bin/bash
+set -eou pipefail
+IFS=$'\n\t'
+
+# these flighting values should match Makefile
+readonly TEST_SUBS="c3dfd792-49a4-4b06-80fc-6fc6d06c4742"
+readonly TEST_REGION="Brazil South"
+
+# supported images (add/update them as new major versions come out)
+readonly DISTROS=(
+	"2b171e93f07c4903bcad35bda10acf22__CoreOS-Beta-877.1.0" \
+	"5112500ae3b842c8b9c604889f8753c3__OpenLogic-CentOS-71-20150731" \
+	"b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-15_04-amd64-server-20151201-en-us-30GB" \
+	"b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_3-LTS-amd64-server-20151117-en-us-30GB" )
+
+# Test constants
+readonly SCRIPT_DIR=$(dirname $0)
+readonly CONCURRENCY=10
+readonly DOCKER_CERTS_DIR=dockercerts
+readonly VM_PREFIX=dockerexttest-
+readonly VM_USER=azureuser
+readonly EXTENSION_NAME=DockerExtension
+readonly EXTENSION_PUBLISHER=Microsoft.Azure.Extensions
+readonly EXTENSION_VERSION=1.0
+readonly EXTENSION_CONFIG=extensionconfig/public.json
+readonly EXTENSION_CONFIG_PROT=extensionconfig/protected.json
+expected_extension_version=
+
+
+### Functions
+
+log() {
+	echo "[$(date +%T)][DEBUG]" "$@"
+}
+
+err() {
+	echo >&2 "[$(date +%T)][ERROR]" "$@"
+}
+
+is_empty() {
```

Comment:
@jasonzio I can see that, I was just recently looking at this article http://www.kfirlavi.com/blog/2012/11/14/defensive-bash-programming/ and trying to see out if I could replace these bash-isms with more human-readable statements for mortals who can't read bash effectively like myself.

I used these only once or twice, I'll fix.